vignettes/shared_cache.Rmd

```diff
@@ -8,45 +12,154 @@ vignette: >
+* <build>**Build status file**</build>: git-versioned files that represent the collective project build status (e.g. which targets have been built and/or are up-to-date) 
+* <target>**Target**</target>: points or steps in the analysis that can be files or R objects. Many of the targets in a shared cache project will be <ind>indicator files</ind>. 
 
-### Pros and Cons
+####Concept
 
-Advantages of a shared cache:
-* Not every analyst needs to build every target, saving on total processing time.
-* Targets that can only be built on specific operating systems (e.g., Mac) or in specific computing environments (e.g., a cluster) can still be accessible to all analysts for further analysis.
-* Intermediate and final products can be immediately visible to anyone who has access to the shared cache, whether they are contributing to the analysis or simply inspecting/using the output.
+A **s**hared **c**ache (**sc** of `scmake`) is a cloud data storage location where raw, intermediate, and/or final data products from an analysis project are contributed to and accessible by multiple analysts. Not all scipiper projects will use a shared cache.
+
+<data>**Data files**</data> only _need_ to be local when the analyst is computing with the <data>**data file**</data>. <ind>**Indicator files**</ind> (`.ind`) represent the remote shared cache among project participants. This allows analyst #1 to compute part of the project, upload output in the project's shared cache, and analyst #2 can use the output from analyst #1 without redoing the computing performed by analyst #1 
+
+Workflow dependencies are connected via the <ind>**indicator files**</ind>. Recipes (e.g. R scripts) used to create the <ind>**indicator files**</ind> pull down the <data>**data files**</data> from the shared cache using `scipiper` functions such as `gd_get()`. 
+
+<build>**Build files**</build> tell remake if the <ind>**indicator files**</ind> (and other <target>**targets**</target> that are <data>**data files**</data>) and their dependencies are up-to-date. <target>**Targets**</target> that are not saved as <ind>**indicator**</ind> or <data>**data**</data> files (e.g. R objects) will not have a <build>**build status file**</build> and will be excuted every time it is called.  
+
+Example of `.ind` file dependency where the function `select_sites()` uses `gd_get()` and the <ind>**indicator file**</ind> `1_data/out/compiled_data.rds.ind` as a dependency to pull down the <data>**data file**</data> (`compiled_data.rds`) from the shared cache. 
+<pre>
+target_default: <target>1_data</target>
+
+sources: 
+  - 1_data/src/gather_data.R
+  - 1_data/src/select_sites.R
+
+targets:
+
+  <target>1_data:</target>
+     depends:
+      - <ind>1_data/out/compiled_data.rds.ind</ind>
+      - <ind>1_data/out/selected_sites.rds.ind</ind>
+
+  <ind>1_data/out/compiled_data.rds.ind:</ind>
+    command: gather_stream_data(
```

Comment:
Could rename to something like `gather_and_share_stream_data` to clarify that pushing to GD happens here.

Reply:
And actually, would you be up for also creating code snippets for `gather_[and_share]_stream_data` and `select_sites` that include the `gd_put` and `gd_get` calls? They're a useful part of the overall SC pattern, I think.

Reply:
Yes, I'll add some code snippets of those functions that get called in the targets